contracts/strategies/looksrare/LooksRareStakingStrategy.sol

```diff
@@ -0,0 +1,244 @@
+    _setPausedInvesting(true);
+  }
+
+  /*
+  *   Resumes the ability to invest into the underlying reward pools
+  */
+  function continueInvesting() public onlyGovernance {
+    _setPausedInvesting(false);
+  }
+
+  function _liquidateReward() internal {
+    uint256 rewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+    if (!sell() || rewardBalance < sellFloor()) {
+      // Profits can be disabled for possible simplified and rapid exit
+      emit ProfitsNotCollected(sell(), rewardBalance < sellFloor());
+      return;
+    }
+
+    notifyProfitInRewardToken(rewardBalance);
+    uint256 remainingRewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+
+    if (remainingRewardBalance <= 0) {
+      return;
+    }
+
+    IERC20(rewardToken()).safeApprove(universalLiquidator(), 0);
+    IERC20(rewardToken()).safeApprove(universalLiquidator(), remainingRewardBalance);
+
+    // we can accept 1 as minimum because this is called only by a trusted role
+    ILiquidator(universalLiquidator()).swapTokenOnMultipleDEXes(
+      remainingRewardBalance,
+      1,
+      address(this), // target
+      storedLiquidationDexes[rewardToken()][underlying()],
+      storedLiquidationPaths[rewardToken()][underlying()]
+    );
+  }
+
+  /*
+  *   Stakes everything the strategy holds into the reward pool
+  */
+  function investAllUnderlying() internal onlyNotPausedInvesting {
+    if(IERC20(underlying()).balanceOf(address(this)) > 0) {
+      enterRewardPool();
+    }
+  }
+
+  /*
+  *   Withdraws all the asset to the vault
+  */
+  function withdrawAllToVault() public restricted {
+    if (address(rewardPool()) != address(0)) {
+      exitRewardPool();
+    }
+    _liquidateReward();
+    IERC20(underlying()).safeTransfer(vault(), IERC20(underlying()).balanceOf(address(this)));
+  }
+
+  /*
+  *   Withdraws the amount for the asset to the vault
+  */
+  function withdrawToVault(uint256 amount) public restricted {
+    // Typically there wouldn't be any amount here
+    // however, it is possible because of the emergencyExit
+    uint256 entireBalance = IERC20(underlying()).balanceOf(address(this));
+
+    if(amount > entireBalance){
+      // While we have the check above, we still using SafeMath below
+      // for the peace of mind (in case something gets changed in between)
+      uint256 needToWithdrawLOOKS = amount.sub(entireBalance);
+
+      uint256 canWithdrawLOOKS = Math.min(rewardPoolBalance(), needToWithdrawLOOKS);
+
+      // due to precision issues between shares -> LOOKS we might end up with getting a tiny amount
+      // of LOOKS less then requested. To ensure we have enough LOOKS after the withdrawal to cover
+      // the requested withdrawal amount, we add a small tolerance. 
+      // the exceeding balance will stay in the strategy until the next action such as hardWork or another withdrawal
+      uint256 tolerance = 1e7;
+      uint256 looksToWithdrawWithTolerance = canWithdrawLOOKS.add(tolerance);
```

Comment:
It seems that this rounding error could be accumulating in the strategy. We prefer flipping the mechanic to favour the strategy, so that the user withdrawing incurs the small rounding error instead. Could you remove the adjustment for the rounding error?

Reply:
Underlying sitting in the strategy would just be invested with the next doHardWork, right?

Reply:
Yes as far as I understand this is a tiny amount that is sitting around in the strategy until the next `doHardWork`? 

I actually proposed this because I thought it ensures better compatibility with the upper level architecture, please see our discussion here (at the first comment of @CryptJS13 in this PR): https://github.com/harvest-finance/harvest-strategy/pull/65#discussion_r789527912

Reply:
Please let me know if I should adjust after all though

Reply:
imagine that  sharesToWithdraw (if tolerance was excluded) would lead to withdrawing less LOOKS tokens than amount.
for example, if amount is 5 LOOKS, but the strategy would have withdrawn 4.95 looks if we just called calculateSharePriceInLOOKS  times canWithdrawLOOKS . The user is entitled to only 4.95 LOOKS in this case. Not 5 LOOKs.
If we include the tolerance, the amount withdrawn would be 5.02 LOOKS (for example). And we send 5 LOOKS to the user. Even if 0.02 would be re-invested later, it turns out that we took away 0.05 LOOKS from other people. So, we socialized the calculation error or discrepancy. And the user was only entitled to 4.95 LOOKS.

Reply:
after talking to @CryptJS13 we will check again internally and let you know

Reply:
Thank you, we discussed internally and while the error seems to be small, we still prefer removing the correction so that no socialization of the losses happens (regardless of how small they are). Can you please do it and re-deploy?